docs/notes/2.21.x.md

```diff
@@ -0,0 +1,153 @@
+
+The default version for many subsystems/backends has been updated:
+
+- [`coverage-py`](https://www.pantsbuild.org/2.21/reference/subsystems/coverage-py): 7.2.7
+- [`debugpy`](https://www.pantsbuild.org/2.21/reference/subsystems/debugpy): 1.6.7.post1
+- [`pants.backend.python.lint.autoflake`](https://www.pantsbuild.org/2.21/reference/subsystems/autoflake): 2.1.1
+- [`pants.backend.python.lint.bandit`](https://www.pantsbuild.org/2.21/reference/subsystems/bandit): 1.7.5
+- [`pants.backend.python.lint.black`](https://www.pantsbuild.org/2.21/reference/subsystems/black): 23.3.0
+- [`pants.backend.python.typecheck.mypy`](https://www.pantsbuild.org/2.21/reference/subsystems/mypy): 1.4.1
+- [`pants.backend.python.lint.pyupgrade`](https://www.pantsbuild.org/2.21/reference/subsystems/pyupgrade): 3.3.2
+- [`pants.backend.python.lint.yapf`](https://www.pantsbuild.org/2.21/reference/subsystems/yapf): 0.40.2
+
+The deprecation has expired for the `python_awslambda` target alias: use the [new `python_aws_lambda_function` target](https://www.pantsbuild.org/2.21/reference/targets/python_aws_lambda_function) instead.
+
+#### Shell
+
+[The `pants.backend.shell.lint.shellcheck` backend](https://www.pantsbuild.org/2.21/reference/subsystems/shellcheck) now uses [version 0.10.0 by default](https://www.pantsbuild.org/2.21/reference/subsystems/shellcheck#version).
+
+Failing `shunit2_test` and `experimental_test_shell_command` tests can now be automatically retried by setting [the `[test].attempts_default` option](https://www.pantsbuild.org/2.21/docs/shell#retries).
+
+#### Terraform
+
+The `experimental-deploy` goal now has built-in support for [the `--dry-run` option](https://www.pantsbuild.org/2.21/reference/goals/experimental-deploy#dry_run), which runs `terraform plan` instead of `terraform apply`.
+
+[The `terraform-hcl2-parser` subsystem](https://www.pantsbuild.org/2.21/reference/subsystems/terraform-hcl2-parser) now uses `python-hcl2` version 4.3.2 by default.
+
+#### Yaml
+
+[The `pants.backend.experimental.tools.yamllint` backend](https://www.pantsbuild.org/2.21/reference/subsystems/yamllint) now uses version 1.32.0 by default.
+
+### Plugin API changes
+
+Plugins that define target generators should set the `generated_target_cls` class var to the targets that they generate. This ensures that `__defaults__` set for that `generated_target_cls` class are applied to the targets generated by the plugin. ([#20649](https://github.com/pantsbuild/pants/pull/20649))
+
+[The new `pants.backend.project_info.peek.HasAdditionalTargetDataFieldSet` union](https://www.pantsbuild.org/2.21/docs/writing-plugins/the-target-api/concepts#adding-information-to-pants-peek-output) allows attaching information to targets that is shown in the `peek` goal, using `pants peek --include-additional-info ...`.
```

Comment:
```suggestion
[The new `pants.backend.project_info.peek.HasAdditionalTargetDataFieldSet` union](https://www.pantsbuild.org/2.21/docs/writing-plugins/the-target-api/concepts#adding-information-to-pants-peek-output) allows attaching information to targets that are shown in the `peek` goal, using `pants peek --include-additional-info ...`.
```

Reply:
Oh, good point. This is ambiguous, the intention was it's like: 

- original: "the new ... union allows attaching information ... that is shown in ..."
- suggestion: "the new ... union allows attaching information ... that are shown in ..."

But I can see it could read as talking about the targets being shown in the goal.

```suggestion
[The new `pants.backend.project_info.peek.HasAdditionalTargetDataFieldSet` union](https://www.pantsbuild.org/2.21/docs/writing-plugins/the-target-api/concepts#adding-information-to-pants-peek-output) allows attaching arbitrary information to targets. This is shown in the `peek` output for these targets by using `pants peek --include-additional-info ...`.
```